java/src/jmri/jmrix/xpa/XpaThrottle.java

```diff
@@ -26,7 +27,7 @@
     public XpaThrottle(LocoAddress address, XpaTrafficController t) {
         super(null);
         this.address = address.getNumber();
-        this.speedIncrement = 1;
+        this.speedStepMode = SpeedStepMode.NMRA_DCC_128;
```

Comment:
It seemed like this was closest to the underlying behavior of the throttle. What do you suggest I do instead?

Reply:
I found some of the documentation about the Xpressnet Phone Adapter. Seems like a weird bit of hardware! It just has "faster", "slower" and "stop" commands, and doesn't expose the absolute speed or limits.

It looks like the speed step mode here doesn't match any of the other speed step modes. Do you think I should add an INCREMENTAL speed step mode for this throttle and maybe the other throttles like it?

Reply:
> I found some of the documentation about the Xpressnet Phone Adapter. Seems like a weird bit of hardware! It just has "faster", "slower" and "stop" commands, and doesn't expose the absolute speed or limits.
> 

The XPA itself was designed to use touch tone telephones as as a throttle when Lenz figured out how expensive it would be to certify a wireless throttle for worldwide use.  It leveraged the fact that touchtone phones were common at the time and cheap locally certified phones were available worldwide.  The idea is the same one that makes WiThrottle popular today, but it was based on technology that is now passé, if and largely obsolete. ( with many people no longer having home phone lines, the once common touch tone phone is less common ).

We use the XPA with a Hayes compatible modem, but all we are really doing is triggering touch tones.

There actually are multiple faster and slower commands but we only use the increment/decrement by 1.

I keep an XPA and a modem around to test with, but it never has been an extremely viable interface solution.  Dialing takes a while, so interaction with locomotives can be slow.  ( turnouts are more responsive because the dial strings are shorter in length. )

> It looks like the speed step mode here doesn't match any of the other speed step modes. Do you think I should add an INCREMENTAL speed step mode for this throttle and maybe the other throttles like it?

There is actually a stopping point, in that when you get to a certain point, the XPA stops incrementing the steps sent to the command station, but there is no feedback that tells us where that point is ( it depends on settings stored in the command station that the XPA does read... it just has no way to inform us or set the number of steps.)

Reply:
Unknown is more appropriate than incremental I think

Reply:
Yeah. I was reading through the Xpa code, and how it mostly sends the ATDT codes to the modem to emulate dialing. It's clever, in a Rube Goldberg sort of way. I can definitely see that it would have its place.

Unknown has a speed increment of 0, and mostly exists to represent when the speed step mode hasn't been initialized at all. This throttle previously had a speed increment of 1. I'm not quite certain, but I think this behaves differently and should be preserved, hence the need for a new speed step mode for this throttle.

Reply:
I didn't see another way to do this, so I implemented an incremental speed step mode for this throttle which emulates the previous behavior.